paasta_tools/metrics/metrics_lib.py

```diff
@@ -24,13 +25,24 @@
 
 
 class TimerProtocol(Protocol):
+    def __enter__(self) -> "TimerProtocol":
```

Comment:
just curious: what are these changes for? doesn't look like we're using any context managers in this diff

Reply:
I was originally using `yelp_meteorite.create_timer` through the metrics_lib abstraction as a context manager, since meteorite allows it, but paasta's metrics_lib stuff doesn't support it so I just added the necessary stuff so that `NoMetrics` could also do it. However, as I commented in m-f-d, I ended up scrapping the context manager usage since on exceptions, meteorite's `create_timer` does not emit. Therefore, this code is optional but it's helpful so I didn't remove it. 

If you have strong feelings about it, I'll remove it.

Reply:
nah, saves us some trouble in the future if we do ever use this in a place that won't potentially throw